src/dispatching/dialogue/storage/serializer.rs

```diff
@@ -0,0 +1,53 @@
+use serde::{de::DeserializeOwned, ser::Serialize};
+use thiserror::Error;
+use Serializer::*;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("failed parsing/serializing JSON: {0}")]
+    JSONError(#[from] serde_json::Error),
+    #[cfg(feature = "cbor-serializer")]
+    #[error("failed parsing/serializing CBOR: {0}")]
+    CBORError(#[from] serde_cbor::Error),
+    #[cfg(feature = "bincode-serializer")]
+    #[error("failed parsing/serializing Bincode: {0}")]
+    BincodeError(#[from] bincode::Error),
+}
+
+type Result<T, E = Error> = std::result::Result<T, E>;
+
+pub enum Serializer {
```

Comment:
Maybe `trait Serializer`? It would be more logical to have in Storage type `T: Serializer`. And it also allow user to put custom serializer.